migration.rst

```diff
@@ -0,0 +1,134 @@
+   single: Migration
+
+Migrating an existing application to Symfony
+============================================
+
+When you have an existing application that was not built with Symfony,
+you might want to move over parts of that application without rewriting
+the existing logic completely. For those cases there is a pattern called
+`Strangler Application`_. The basic idea of this pattern is to create a
+new application that gradually takes over functionality from an existing
+application. This migration approach can be implemented with Symfony in
+various ways and has some benefits over a rewrite such as being able
+to introduce new features in the existing application and reducing risk
+by avoiding a "big bang"-release for the new application.
+
+Prerequisites
+-------------
+
+The following steps do not require you to have the new Symfony
+application in place and in fact it might be safer to introduce these
+changes beforehand in your existing application.
+
+Before you start introducing Symfony to the existing application, you
+have to ensure certain requirements are met by your existing application.
+By extension this means that you will have to decide which version you
+are aiming to migrate to, either a current stable release or the long
+term support version (LTS). The main difference is, how frequently you
+will need to upgrade in order to use a supported version. Using the
+most recent, stable release will likely give you more features, but it
+will also require you to update more frequently to ensure you will get
+support for bug fixes and security patches and you will have to work
+faster on fixing deprecations to be able to upgrade.
+
+First of all your environment needs to be able to support the minimum
+requirements for both applications. In other words, when the Symfony
+release you aim to use requires PHP 7.1 and your existing application
+does not yet support this PHP version, you will probably have to upgrade
+your legacy project. You can find out the
+:doc:`requirements </reference/requirements>` for running Symfony and
+compare them with your current application's environment to make sure you
+are able to run both applications on the same system. Having a test
+system, that is as close to the production environment as possible,
+where you can just install a new Symfony project next to the existing one
+and check if it is working will give you an even more reliable result.
+
+.. tip::
+
+    If your current project is running on an older PHP version such as
+    PHP 5.x upgrading to a recent version will give you a performance
+    boost without having to change your code.
+
+In older PHP applications it was quite common to rely on global state and
+even mutate it during runtime. This might have side effects on the newly
+introduced Symfony application. In other words code relying on super
+globals like ``$_POST`` in the existing application should be refactored,
+to allow for both systems to work simultaneously.
```

Comment:
Superglobals aren't that much of a problem, actually. You can have Symfony 4 controllers and services that access the superglobals.

However the usage of regular global variables causes trouble if you attempt to have Symfony route requests into your legacy application. If you take a piece of code that is intended to run in the context of the global variable scope and wrap it into a function, it'll probably break.